### DISCUSSION TITLE
Make `AudioStreamRandomizer` `random_pitch` perceptually uniform.

### DISCUSSION THREAD

**Discussion by ettiSurreal** [godotengine/godot-proposals#10238] [General Discussions]
A recent tweet from Freya Holmér has caught my attention.
https://x.com/FreyaHolmer/status/1813629237187817600
Quote:
> unsolicited fun fact!! since pitch is a multiplicative quantity, uniformly distributed random values will be biased toward the higher pitch
to make it *perceptually* uniform, you can randomize in log scale:
`pitch = a*exp(t*ln(b/a))`
where:
`t = Random.value (0 to 1)`
`a = 0.8, b = 1.2`

Example from another user:
> Just to provide an example for this, if you go from a C an octave up to another C, you go from a pitch of 1 to 2. 
If you instead go an octave down, you go to 0.5 instead.

Looking through the `AudioStreamRandomizer` code it seems to just take the naive approach, so, thoughts on taking this approach instead? Would performance be of concern (maybe add a bool toggle for it if so)?

**Comment by Calinou** [comment]
Feel free to open a pull request for this 🙂 

> Would performance be of concern (maybe add a bool toggle for it if so)?

I'd imagine the cost of this to be pretty small in real world use cases, considering you might have 20 sounds with random pitch starting playback per second in a worst-case scenario. The CPU cost of decoding samples (even MP3) is far higher in comparison.
